### PR TITLE
fix(compiler): throw descriptive error meesage for invalid NgModule providers

### LINKED ISSUE #10714: NgModule: better error message than "Error: Invalid provider ... got: [object Object]"
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report
[x] feature request
[ ] support request 
```
I got the error
```
(index):20 Error: Error: Invalid provider - only instances of Provider and Type are allowed, got: [object Object](…)
(anonymous function) @ (index):20
ZoneDelegate.invoke @ zone.js:323
Zone.run @ zone.js:216
(anonymous function) @ zone.js:571
ZoneDelegate.invokeTask @ zone.js:356
Zone.runTask @ zone.js:256
drainMicroTaskQueue @ zone.js:474
ZoneTask.invoke @ zone.js:426
```
Do you know what the error is or where to find it?

Fortunately , my app is small so I started at the top.
Eventually I came to this.. Can you see the error?
```
@NgModule({
  imports:      [ SharedModule ],
  declarations: [ DashboardComponent ],
  providers:    [ RouterModule.forChild(routes) ]
})
```

Answer: The ` RouterModule.forChild(routes)` belongs in `imports` not `providers`.

**Current behavior**

Mysterious error with no clue as to where it is or what the offending thing is. I can tell it has something to do with providers ... but what and where?

**Expected/desired behavior**

Some indication of the file that has the problem or the the class (here `AppModule`). Would be really nice if it could say something other than `[Object, Object]`

**Reproduction of the problem**

Pick an app (e.g., the sample in "Angular Modules" chapter), find a feature module, and move the ` RouterModule.forChild(routes)` from `imports` to `providers`.  Give it to someone else and ask that person to find the bug.

**Please tell us about your environment:**

* **Angular version:** 2.0.0-rc.5
 
* **Browser:** [ all  ] 
 
* **Language:** [ TS ]

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
Fixes #10714

### CODE REVIEW

File: modules/@angular/core/test/linker/ng_module_integration_spec.ts [hunk 1]
```diff
@@ -1171,6 +1177,20 @@ function declareTests({useJit}: {useJit: boolean}) {
...
+          class ImportedModule1 {
+          }
+
+          @NgModule({imports: [{ngModule: ImportedModule1, providers: [<any>'broken']}]})
+          class SomeModule {
+          }
+
+          expect(() => createModule(SomeModule).injector)
+              .toThrowError(
+                  `Invalid provider for the module 'ImportedModule1' - only instances of Provider and Type are allowed, got: [?broken?]`);
```
Comment (IgorMinar) [inline modules/@angular/core/test/linker/ng_module_integration_spec.ts]: can you change "module" to "NgModule"? we should be about what module we mean. thanks!

**Comment by IgorMinar** [comment]
otherwise lgtm

**Comment by IgorMinar** [comment]
once you correct this, please remove cleanup label and add "pr_action: merge" label

**Comment by pkozlowski-opensource** [comment]
Cleaned up, squashed & rebased. This one should be good to go.

**Comment by kara** [comment]
@pkozlowski-opensource  Sorry, but can you rebase again? :D

**Comment by pkozlowski-opensource** [comment]
@kara done. All back to green.

**Pull request merged**